src/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs

```diff
@@ -32,19 +32,29 @@ public void TestEnvironmentProperty()
 
             IDictionary<string, string> environment = psi.Environment;
 
-            Assert.NotEqual(environment.Count, 0);
+            Assert.NotEqual(0, environment.Count);
 
-            int CountItems = environment.Count;
+            int countItems = environment.Count;
 
             environment.Add("NewKey", "NewValue");
             environment.Add("NewKey2", "NewValue2");
 
-            Assert.Equal(CountItems + 2, environment.Count);
+            Assert.Equal(countItems + 2, environment.Count);
             environment.Remove("NewKey");
-            Assert.Equal(CountItems + 1, environment.Count);
+            Assert.Equal(countItems + 1, environment.Count);
 
-            //Exception not thrown with invalid key
-            Assert.Throws<ArgumentException>(() => { environment.Add("NewKey2", "NewValue2"); });
+            // The full .NET framework represents Environment with StringDictionary. On .NET Core we use
+            // use Dictionary<string, string>, because we need to have case sensitive behaviour on Unix.
```

Comment:
The full .NET framework has a couple of differences:
- The `environmentVariables` variable is a StringDictionary (vs Dictionary<string, string in corefx)
- The implementation of `EnvrionmentVariables` creates a dictionary from `Environment`. In corefx this is the other way around. 
- The return value of a StringDictionary in terms of IDictionary<string, string> (https://github.com/Microsoft/referencesource/blob/4fe4349175f4c5091d972a7e56ea12012f1e7170/System/compmod/system/collections/specialized/stringdictionary.cs#L188) has a pretty long and convoluted adapter that maintains the behaviour of StringDictionary for IDictionary<string, string>. This internal state is available to ProcessStartInfo in mscorlib but not in corefx.

If we want to unite with mscorlib behaviour I think we'll need to reimplement the entire IDictionary adapter. It's doable but is a bit of a nuiscance.

Then we'll need to make sure that StringDictionaryWrapper has the correct behaviour that matches with netfx. And we need to use Dictionary<string, string> instead of StringDictionary as we want platform dependant case sensitive that StringDictionary doesn't support.